src/imageproxy.rs

```diff
@@ -701,11 +701,16 @@ mod tests {
         };
         config.skopeo_cmd = Some(Command::new("no-skopeo"));
 
-        let res = ImageProxy::new_with_config(config).await;
-        assert!(matches!(res, Err(Error::SkopeoSpawnError(_))));
-        assert_eq!(
-            res.err().unwrap().to_string(),
-            "skopeo spawn error: No such file or directory (os error 2)"
-        );
+        match ImageProxy::new_with_config(config).await {
+            Ok(_) => panic!("Expected an error"),
+            Err(ref e @ Error::SkopeoSpawnError(ref inner)) => {
```

Comment:
Interesting note: I needed to start using `ref` as we want to be able to look at both the inner and outer parts. This is pretty unusual IME.